### PR TITLE
Updated RPM spec and added useful modules

### PR DESCRIPTION
**Pull request by gmmephisto**
- added rhel6/7 and `python3` support in RPM spec
- added `psys.process` module with process initialization and termination signal handling function
- added `psys.pipe` module with UNIX Pipe wrapper
- added `psys.daemon.write_pidfile` function
- refactored `Makefile`

RPMs for testing will be available in [copr repository](https://copr.fedorainfracloud.org/coprs/miushanov/pyaddons/build/180546/).

### CODE REVIEW

File: psys/daemon.py [hunk 1]
```diff
@@ -82,3 +83,46 @@ def release_pidfile(path, fd):
...
+
+
+def write_pidfile(fd):
+    """Write pid to pidfile previously allocated by acquire_pidfile()"""
+
+    data = str(os.getpid())
+    datalen = len(data)
+
+    while data:
+        size = os.write(fd, data)
```
Comment (KonishchevDmitry) [inline psys/daemon.py]: Needs to be wrapped by `eintr_retry()`.

File: psys/daemon.py [hunk 2]
```diff
@@ -82,3 +83,46 @@ def release_pidfile(path, fd):
...
+        data = data[size:]
+    os.ftruncate(fd, datalen)
```
Comment (KonishchevDmitry) [inline psys/daemon.py]: Needs to be wrapped by `eintr_retry()`.

File: psys/daemon.py [hunk 3]
```diff
@@ -82,3 +83,46 @@ def release_pidfile(path, fd):
...
+            if os.fork():
+                os._exit(0)
+
+    os.chdir("/")
+    os.umask(0)
+
+    signal.signal(signal.SIGHUP, signal.SIG_IGN)
+    signal.siginterrupt(signal.SIGHUP, False)
+
+    # Redirecting stdout and stderr to the log file and
```
Comment (KonishchevDmitry) [inline psys/daemon.py]: Misleading comment: actually we don't redirect stdout/stderr to any log file - we redirect them to /dev/null.

**Comment by gmmephisto** [comment]
@KonishchevDmitry, thanks for review! Fixed!

**Pull request merged**